src/plugins/MiradorPDIIIFDialog.js

```diff
@@ -31,97 +33,114 @@ const mapStateToProps = (state, { windowId, containerId }) => ({
+          },
+        ],
+      });
+    } catch (e) {
+      this.setState({ savingError: e.message });
+      console.error(e);
+      return Promise.reject(e);
+    }
+
+    try {
+      if ((await handle.queryPermission({ mode: "readwrite" })) !== "granted") {
+        // Throw error if permission is not granted
+        // N.B. I wasn't able to trigger this error (e.g. by choosing folder with strict permission)
+        // but it's here for completeness
+        throw new Error("Permission to write to file was not granted");
+      } else {
+        // Reset the error state
+        this.setState({ savingError: null });
 
-      try {
-        if (
-          (await handle.queryPermission({ mode: "readwrite" })) !== "granted"
-        ) {
-          // Throw error if permission is not granted
-          // N.B. I wasn't able to trigger this error (e.g. by choosing folder with strict permission)
-          // but it's here for completeness
-          throw new Error("Permission to write to file was not granted");
-        } else {
-          // Reset the error state
-          this.setState({ savingError: null });
-
-          const pdfPath = (await handle.getFile()).name;
-          const webWritable = await handle.createWritable();
-
-          closeDialog();
-
-          // Start the PDF generation
-          return await convertManifest(manifest, webWritable, {
-            concurrency: 4,
-            maxWidth: 1500,
-            coverPageEndpoint: "https://pdiiif.jbaiter.de/api/coverpage",
-          });
-        }
-      } catch (e) {
-        // Display permission / conversion error
-        this.setState({ savingError: e.message });
-        return console.error(e);
+        const pdfPath = (await handle.getFile()).name;
+        const webWritable = await handle.createWritable();
+
+        closeDialog();
+
+        // Start the PDF generation
+        return await this.manifestConverter(webWritable);
       }
+    } catch (e) {
+      // Display permission / conversion error
+      this.setState({ savingError: e.message });
+      console.error(e);
+      return Promise.reject(e);
     }
-  };
+  }
+
+  /**
+   * Downoloads the PDF using the Streamsaver API
+   * @param {string} label
+   * @returns {Promise}
+   */
+  async downloadWithStreamsaver(label) {
+    const { closeDialog } = this.props;
+
+    const webWritable = new streamSaver.createWriteStream(`${label}.pdf`);
+
+    closeDialog();
+
+    // Start the PDF generation
+    return await this.manifestConverter(webWritable);
+  }
+
+  /**
+   * Call to PDIIIF convertManifest shared between download methods
+   * @param {WritableStream} webWritable Stream required by PDIIIF
+   * @returns {Promise}
+   */
+  async manifestConverter(webWritable) {
+    const { manifest } = this.props;
+    const abortController = new AbortController();
+
+    return await convertManifest(manifest.json, webWritable, {
+      maxWidth: 1500,
```

Comment:
`concurrency: 4` needs to be added here.

Reply:
Good spot, this must have gotten lost in my refactor (after I made a deliberate mental note to myself NOT forget to put this back). I'll put this back :)